### ISSUE TITLE
Informer le nom d'utilisateur dans un article sent 

### ISSUE THREAD

**Issue by Mmellet** [EcrituresNumeriques/stylo#593]
C'est un élément dont on s'est rendu compte qu'il serait utile de notifier le nom de l'utilisateur stylo pour un article Sent : 

![Screenshot 2022-03-17 at 17-28-24 Stylo](https://user-images.githubusercontent.com/40469215/158898806-0ff59e36-f188-4192-bbe4-16bebd678d10.png)

Il serait utile, comme pour le cas du Share, que le nom de l'utilisateur qui a envoyé l'article apparaisse dans le profil de l'article (en dessous de son titre, où sont listé les comptes stylo qui ont accès à l'article. 

Pour le cas d'utilisation revue, comme les auteurs nous envoient parfois des articles (sans mentionner leur nom ni leur mail dans l'article), on pourrait aussi imaginer que l'adresse mail du compte utilisateur qui a envoyé l'article apparaisse au passage du curseur. Cela reste des propositions !

**Comment by thom4parisot** [comment]
Aujourd'hui le comportement du "Sent" c'est que la personne destinataire devient "propriétaire" d'une copie du document original. La personne émettrice conserve son document original indépendamment.

Est-ce que la personne émettrice est censée pouvoir modifier cette copie ?

Ou ça se résume "juste" à dire de qui _provient_ l'article ?

**Comment by Mmellet** [comment]
Non le comportement est le bon ainsi, juste informer de qui provient l'article

**Comment by thom4parisot** [comment]
Dac, merci pour ta clarification 🙁 

Je me demande aussi : la personne qui "envoie" le document, elle continue à le modifier de votre côté ? Ou d'une certaine manière, elle "passe la main" sur l'écriture ?

Ça nous aidera à décider comment/où on stocke cette information.

**Comment by Mmellet** [comment]
Je pense que ce qui se passe (et qui convient selon moi) c'est que quand la personne envoie un article, son article est en réalité dupliqué et le doublon est envoyé à l'autre utilisateur, qui a alors les pleins droits dessus. Donc oui elle passe la main mais une duplication de son article en réalité. Le seul problème actuellement c'est que l'utilisateur ne sait pas qui lui a envoyé un article.

**Comment by ggrossetie** [comment]
Est-ce que si on ajoute cette information dans le titre c'est acceptable ? Par exemple : "[Sent by mmellet] Pour en finir avec le mal...".

Le désavantage de cette solution : 
- Si on change le titre, on perd l'information
- L'information est mélangée dans le titre. Autrement dit, il n'y a pas un moyen fiable d'exploiter cette information. Par exemple, on aurait pu imaginer regrouper l'ensemble des articles envoyés par utilisateur. Tu aurais pu alors filtrer ta liste des articles pour voir uniquement les articles envoyés par Thomas ou Maïtané. 

Si on veut stocker cette information à part, alors on peut soit l'utilisateur dans `contributors` avec un role `concede` (pour indiquer qu'il a cédé son article) ? Ou alors un nouveau champ qui pourra contenir des informations sur l'action d'envoyer une copie à un autre utilisateur :

```js
{
  sendCopy: {
    date: "2022-03-18T06:00:00Z" 
    from: ObjectId("1234")
  }
}
```

**Comment by thom4parisot** [comment]
Oui, j'essayais de voir si ça pouvait entrer dans le champ `contributors`.

Je trouverais même intéressant d'aller vers la suppression du champ `owner`, et d'avoir cette information sous forme d'un rôle `originalAuthor` dans le champ `contributors.x.role`.

Ça nous permettrait peut-être de faire un premier pas vers la notion d'archivage (un rôle en plus ou en moins).
